---
layout: reference
title: matador::unit_exception Class Reference
---
<!-- Generated by Doxygen 1.8.14 -->
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',false,false,'search.php','Search');
});
/* @license-end */</script>
<div id="main-nav"></div>
<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespacematador.html">matador</a></li><li class="navelem"><a class="el" href="classmatador_1_1unit__exception.html">unit_exception</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="classmatador_1_1unit__exception-members.html">List of all members</a> &#124;
<a href="#pub-methods">Public Member Functions</a>  </div>
  <div class="headertitle">
<div class="title">matador::unit_exception Class Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>Unit exception class thrown on a unit test assert definition.  
 <a href="classmatador_1_1unit__exception.html#details">More...</a></p>

<p><code>#include &lt;unit_exception.hpp&gt;</code></p>

<p>Inherits std::exception.</p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a4aa6128812e5496a4673e00099dc3a33"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmatador_1_1unit__exception.html#a4aa6128812e5496a4673e00099dc3a33">unit_exception</a> (std::string msg)  throw ()</td></tr>
<tr class="memdesc:a4aa6128812e5496a4673e00099dc3a33"><td class="mdescLeft">&#160;</td><td class="mdescRight">Contructs a <a class="el" href="classmatador_1_1unit__exception.html" title="Unit exception class thrown on a unit test assert definition. ">unit_exception</a>.  <a href="#a4aa6128812e5496a4673e00099dc3a33">More...</a><br /></td></tr>
<tr class="separator:a4aa6128812e5496a4673e00099dc3a33"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afeebbfb1c48187b02f890dc6f534e8a6"><td class="memItemLeft" align="right" valign="top">virtual const char *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmatador_1_1unit__exception.html#afeebbfb1c48187b02f890dc6f534e8a6">what</a> () const  throw ()</td></tr>
<tr class="separator:afeebbfb1c48187b02f890dc6f534e8a6"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Unit exception class thrown on a unit test assert definition. </p>
<p>This <a class="el" href="classmatador_1_1unit__exception.html" title="Unit exception class thrown on a unit test assert definition. ">unit_exception</a> is thrown when an assert evaluates to false. This exception is caugt by the main <a class="el" href="classmatador_1_1test__suite.html" title="The container for all unit tests. ">test_suite</a> serializable. </p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a id="a4aa6128812e5496a4673e00099dc3a33"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a4aa6128812e5496a4673e00099dc3a33">&#9670;&nbsp;</a></span>unit_exception()</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">matador::unit_exception::unit_exception </td>
          <td>(</td>
          <td class="paramtype">std::string&#160;</td>
          <td class="paramname"><em>msg</em></td><td>)</td>
          <td></td>
        </tr>
        <tr>
          <td align="right">throw </td><td>(</td><td colspan="2"></td>
        </tr>
        <tr>
          <td align="right"></td><td>)</td><td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">explicit</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Contructs a <a class="el" href="classmatador_1_1unit__exception.html" title="Unit exception class thrown on a unit test assert definition. ">unit_exception</a>. </p>
<p>A serializable of type <a class="el" href="classmatador_1_1unit__exception.html" title="Unit exception class thrown on a unit test assert definition. ">unit_exception</a> with the given message is created.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">msg</td><td>Message of the exception. </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a id="afeebbfb1c48187b02f890dc6f534e8a6"></a>
<h2 class="memtitle"><span class="permalink"><a href="#afeebbfb1c48187b02f890dc6f534e8a6">&#9670;&nbsp;</a></span>what()</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">virtual const char* matador::unit_exception::what </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
        <tr>
          <td align="right">throw </td><td>(</td><td colspan="2"></td>
        </tr>
        <tr>
          <td align="right"></td><td>)</td><td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns the message of this exception.</p>
<dl class="section return"><dt>Returns</dt><dd>The message of the exception. </dd></dl>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>matador/unit/unit_exception.hpp</li>
</ul>
</div><!-- contents -->
